packages/@react-stately/combobox/src/useComboBoxState.ts

```diff
@@ -306,12 +306,26 @@ export function useComboBoxState<T extends object>(props: ComboBoxStateOptions<T
     setFocusedState(isFocused);
   };
 
+  let selectionManagerProxy = useMemo(() => {
```

Comment:
Kinda gross, still digging if there is an alternative way. I don't think there is though since clicking a option in the ListBox dropdown triggers the selectionManager's [replaceSelection](https://github.com/adobe/react-spectrum/blob/c53cfb204f6977e1ac620fdfa841ffdcf567c5f0/packages/%40react-aria/selection/src/useSelectableItem.ts#L124), which looks like [this](https://github.com/adobe/react-spectrum/blob/c53cfb204f6977e1ac620fdfa841ffdcf567c5f0/packages/%40react-stately/selection/src/SelectionManager.ts#L337-L352). The `this.state.setSelectedKeys(selection);` is calling `setSelectedKeys` from `useMultipleSelectionState` which we can't directly interfere with from `useComboBoxState`

Alternative is to make the ComboBox's ListBox `shouldSelectOnPressUp` to be false, feels iffy though to change that behavior just to fix this bug though

Reply:
Some other ideas:
- open up SelectionManager so we can change `selectionMode` to 'none' when `isOpen` is `false`
  -  not great, more of a bandaid
- "freeze" the collection when the menu is closed
  - we could either return the ability to freeze/unfreeze the collection from useComboBoxState and allow the user to control when to freeze/unfreeze. Feels like potentially unwanted complexity
  - we could handle the freezing ourselves, but would need to introduce a prop so the user could tell us when to unfreeze it (aka when their menu transition is done)

Reply:
I think we need a new prop to the selectionManager, using the Proxy is clever, but the selectionManager should be able to support all of its usecases within hooks

what is it that we are doing, are we preventing all changes to the selection? or just replace? what if the selectionMode was multiple or the interaction mode was such that clicking added to the selection? Obviously that can't happen in a combobox, but maybe something else could.

Reply:
So in a general case, I'd imagine we would want to prevent all changes to selection (thus something like changing change `selectionMode` to 'none' when `isOpen` is `false` would be something like this, but perhaps a different property should be made like `freezeSelection`). I am a bit wary about introducing something brand new to SelectionManager in this case though for a couple of reasons though:

1. This bug that we are solving for is quite specific since it is a blur on click that also happens to mutate the listbox in such a way that it might intersect with your click position as it transitions out. I'd rather not introduce a brand new control to SelectionManager without a more general use case/need for it.
2. If we are ok with adding a broader control to prevent selection changes, then SelectionManager may not be the best place/only place to add it since the call to `state.setSelectedKeys` is really the thing that is changing selection and thus SelectionManager isn't the only place where a call to that could happen. The change would then need to happen in places like `useMultipleSelectionState` where we would add another conditional [here](https://github.com/adobe/react-spectrum/blob/97ff9f95d91befaf87251e52ea484f81daae8f3a/packages/%40react-stately/selection/src/useMultipleSelectionState.ts#L113)

Reply:
Good points, I guess this proxy doesn't prevent us from fixing elsewhere either and it addresses the immediate issue.